homework24/tests/test_browser.py

```diff
@@ -0,0 +1,117 @@
+    wait = WebDriverWait(browser, 10)
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, ContactListPageLocators.ADD_A_NEW_CONTACT_BUTTON))).click())
+    for field, value in contact_info.items():
+        element = wait.until(EC.presence_of_element_located((By.XPATH, field)))
+        element.send_keys(value)
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, AddContactPageLocators.SUBMIT_BUTTON))).click())
+
+
+def edit_contact(browser, updated_info):
+    """Function edit a contact."""
+    wait = WebDriverWait(browser, 10)
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, EditContactPageLocators.CONTACT_TABLE_ROW))).click())
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, EditContactPageLocators.EDIT_CONTACT_BUTTON))).click())
+    time.sleep(1)
+    for field, value in updated_info.items():
+        element = wait.until(EC.element_to_be_clickable((By.XPATH, field)))
+        element.clear()
+        element.send_keys(value)
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, EditContactPageLocators.SUBMIT_BUTTON))).click())
+    time.sleep(5)
+
+
+def delete_contact(browser):
+    """Function delete a contact."""
+    wait = WebDriverWait(browser, 10)
+    contact_row = (wait.until
+                   (EC.presence_of_element_located
+                    ((By.XPATH, DeleteContactDetails.CONTACT_TABLE_ROW))))
+    contact_name = contact_row.text
+    contact_row.click()
+    (wait.until
+     (EC.element_to_be_clickable
+      ((By.XPATH, DeleteContactDetails.DELETE_CONTACT_BUTTON))).click())
+    wait.until(EC.alert_is_present()).accept()
+    try:
+        (wait.until
+         (EC.presence_of_element_located
+          ((By.XPATH, f"//*[contains(text(), '{contact_name}')]"))))
+        return False
+    except TimeoutException:
+        return True
+
+
+def test_add_contact(open_browser):
+    """Test case for adding a new contact."""
+    contact_info = {
+        AddContactPageLocators.ADD_FIRSTNAME_INPUT: 'Kate',
+        AddContactPageLocators.ADD_LAST_NAME_INPUT: 'Mirror',
+        AddContactPageLocators.ADD_DATE_OF_BIRTH_INPUT: '1988-05-09',
+        AddContactPageLocators.ADD_EMAIL_INPUT: 'zxcvb@gmail.com',
+        AddContactPageLocators.ADD_PHONE_INPUT: '2342635332',
+        AddContactPageLocators.ADD_STREET_ADDRESS1_INPUT: 'First st 6',
+        AddContactPageLocators.ADD_STREET_ADDRESS2_INPUT: 'Second 1',
+        AddContactPageLocators.ADD_CITY_INPUT: 'New York',
+        AddContactPageLocators.ADD_STATE_OF_PROVINCE_INPUT: 'NY',
+        AddContactPageLocators.ADD_POSTAL_CODE_INPUT: '123409',
+        AddContactPageLocators.ADD_COUNTRY_INPUT: 'USA',
+    }
+    browser = open_browser
+    add_contact(browser, contact_info)
+    contact_row = (WebDriverWait
+                   (browser, 10).until
+                   (EC.presence_of_element_located
+                    ((By.XPATH, EditContactPageLocators.CONTACT_TABLE_ROW))))
```

Comment:
it should be on it's page -> edit_contact_page(if I'm not mistaken)

Reply:
edit_contact_page contains locators, can you look again and clarify?

Reply:
my bad